report_custom_filename/controllers/reports.py

```diff
@@ -39,7 +39,14 @@ def index(self, action, token):
         for report in report_xml.browse(report_ids):
             if not report.download_filename:
                 continue
-            objects = http.request.session.model(context['active_model'])\
+            # When you are on the form view of a customer and you
+            # click on the "Sales" button, context['active_model']
+            # is still res.partner (probably a bug...) but
+            # context['active_ids'] has the ID of the sale order
+            # cf https://github.com/OCA/reporting-engine/issues/56
+            # That's why we use report.model instead of context['active_model']
+            # together with context['active_ids']
+            objects = http.request.session.model(report.model)\
                 .browse(context['active_ids'])
```

Comment:
thanks for the ping, I think what you do is the right thing, and it's coincidence that the active_ids thing works very often. Can you change this line too to get the report's object ids? Cf https://github.com/OCA/OCB/blob/8.0/addons/web/controllers/main.py#L1574